capstone-app/src/app/modules/history/wiki-results.service.ts

```diff
@@ -17,15 +19,74 @@ export class WikiResultsService {
     'https://en.wikipedia.org/w/api.php?origin=*&action=parse&page=';
   private static readonly URL_END = '&format=json';
 
-  search(query: string): Observable<WikiSearchResult> {
+  search(query: string): Observable<{title: string, history: string, furtherReading: Map<string, string>}> {
     return this.searchHandler
       .getQueryString(query)
       .pipe(
         switchMap((result) =>
           this.http.get<WikiSearchResult>(
             `${WikiResultsService.URL_BEGINNING}${result}${WikiResultsService.URL_END}`
           )
-        )
+        ),map(history => {
+          if (history.parse.text) {
+            const historyResult = this.fixString(history.parse.text['*']);
+            const title = history.parse.title;
+            return {title, ...historyResult};
+          }
+          else {
+            throw new Error("API did not return a valid response");
+          }
+        }),
+        catchError(() => {
+          console.log('API did not return a valid response.');
+          const title = '';
+          const history = '';
+          const furtherReading = new Map<string, string>();
+          return of({title, history, furtherReading});
+        })
       );
   }
+
+  fixString(text: string): {history: string, furtherReading: Map<string, string>} {
+    const firstIndex = text.indexOf('<span class="mw-headline" id="History">History</span>', 0);
+    if (firstIndex !== -1){
+      const firstPartOfString = text.substring(firstIndex, text.length);
+      const endIndex = firstPartOfString.indexOf('<h2>', 0);
+      const startIndex = firstPartOfString.indexOf('</h2>', 0);
+      let middleOfString = firstPartOfString.substring(startIndex, endIndex);
+      const paragraphs = middleOfString.split('<p>');
+      if (paragraphs.length > 6){
+        middleOfString = paragraphs.slice(0,5).join('');
+      }
+      const furtherReading = this.findHrefs(middleOfString);
+      let history = middleOfString.split(/<.*?>/g).join('');
+      history = history.split(/&.*?;/g).join('');
+      return {history, furtherReading};
+    }
+    console.log('The city page was found, but unfortunately there was no history paragraph found!');
+    let history = text.split(/<.*?>/g).join('');
+    history = history.split(/&.*?;/g).join('');
+    return {history, furtherReading: new Map()};
+  }
+
+  findHrefs(text: string): Map<string, string> {
+    const urls = new Map<string, string>();
+    const el = document.createElement('p');
+    el.innerHTML = text;
+    const hrefs = Array.from(el.querySelectorAll('a'));
+    let count = 1;
+    for (const h of hrefs) {
+      if (h.getAttribute('href').charAt(0) === '#' || (h.getAttribute('title') && h.getAttribute('title').startsWith('Edit'))){
+        delete hrefs[hrefs.indexOf(h)];
+      }
+      else if (count <= 15) {
+        let name = h.getAttribute('title');
+        let url = h.getAttribute('href').toString();
+        url = 'https://en.wikipedia.org' + url;
```

Comment:
Okay one comment here and it's entirely optional as it depends on your vision for your section. Either when the user clicks on the related article, it should 1) Open that article in a new tab or 2) Display some contents of that article on our webpage. Since our goal is to consolidate resources into one location, would 2) make more sense especially as you already have the wiki API setup? And if so, then you just need to store the query string of the url.

Reply:
No, because 
1. Right now I have no way to query a new page with a url.
2. The point of the history section is to be short and concise. If we rendered more info on the page it would still be a short summary. By giving the user a link it is still consolidated in that they did not need to search to find the info, but it also provides a much more in depth view for them.